source/utilities.tex

```diff
@@ -19492,3 +19493,1887 @@
+\pnum
+For any types \tcode{T} and \tcode{charT} for which
+neither the library nor the user provides
+an explicit or partial specialization of
+the class template \tcode{formatter},
+\tcode{formatter<T, charT>} is disabled.
+
+\pnum
+If the library provides an explicit or partial specialization of
+\tcode{formatter<T, charT>}, that specialization is enabled
+except as noted otherwise.
+
+\pnum
+If \tcode{F} is a disabled specialization of \tcode{formatter}, these
+values are \tcode{false}:
+\begin{itemize}
+\item \tcode{is_default_constructible_v<F>},
+\item \tcode{is_copy_constructible_v<F>},
+\item \tcode{is_move_constructible_v<F>},
+\item \tcode{is_copy_assignable_v<F>}, and
+\item \tcode{is_move_assignable_v<F>}.
+\end{itemize}
+
+\pnum
+An enabled specialization \tcode{formatter<T, charT>} meets the
+\newoldconcept{Formatter} requirements\iref{formatter.requirements}.
+\begin{example}
+\begin{codeblock}
+#include <format>
+
+enum color { red, green, blue };
+const char* color_names[] = { "red", "green", "blue" };
+
+template<> struct std::formatter<color> : std::formatter<const char*> {
+  auto format(color c, format_context& ctx) {
+    return formatter<const char*>::format(color_names[c], ctx);
+  }
+};
+
+struct err {};
+
+std::string s0 = std::format("{}", 42);         // OK, library-provided formatter
+std::string s1 = std::format("{}", L"foo");     // ill-formed: disabled formatter
+std::string s2 = std::format("{}", red);        // OK, user-provided formatter
+std::string s3 = std::format("{}", err{});      // ill-formed: disabled formatter
+\end{codeblock}
+\end{example}
+
+\rSec3[formatter.requirements]{Formatter requirements}
+
+\pnum
+\indextext{requirements!\idxnewoldconcept{Formatter}}%
+A type \tcode{F} meets the \newoldconceptdefn{Formatter} requirements if:
+
+\begin{itemize}
+\item
+it meets the
+\begin{itemize}
+\item \oldconcept{DefaultConstructible} (\tref{cpp17.defaultconstructible}),
+\item \oldconcept{CopyConstructible} (\tref{cpp17.copyconstructible}),
+\item \oldconcept{CopyAssignable} (\tref{cpp17.copyassignable}), and
+\item \oldconcept{Destructible} (\tref{cpp17.destructible})
+\end{itemize}
+requirements,
+
+\item
+it is swappable\iref{swappable.requirements} for lvalues, and
+
+\item
+the expressions shown in \tref{formatter} are valid and
+have the indicated semantics.
+\end{itemize}
+
+\pnum
+Given character type \tcode{charT}, output iterator type
+\tcode{Out}, and formatting argument type \tcode{T},
+in \tref{formatter}:
+\begin{itemize}
+\item \tcode{f} is a value of type \tcode{F},
+\item \tcode{u} is an lvalue of type \tcode{T},
+\item \tcode{t} is a value of a type convertible to (possibly const) \tcode{T},
+\item \tcode{PC} is \tcode{basic_format_parse_context<charT>},
+\item \tcode{FC} is \tcode{basic_format_context<Out, charT>},
+\item \tcode{pc} is an lvalue of type \tcode{PC}, and
+\item \tcode{fc} is an lvalue of type \tcode{FC}.
+\end{itemize}
+\tcode{pc.begin()} points to the beginning of the
```

Comment:
Well, it's not given by [format.string]p2, because the wording fails to connect the individual argument formatting steps with invoking particular member functions on particular instances of `formatter<T, charT>`. The intended meaning is obvious to a careful reader who is paying attention, but from an editorial standpoint this is still a hole we should fix. (The point of the FIXME isn't that we don't know the answer, it's that we don't *say*.)

Reply:
This has been resolved, right?